private/bufpkg/bufimage/bufimagemodify/internal/file_options_trie_test.go

```diff
@@ -0,0 +1,342 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFileOptionsTrieInsert(t *testing.T) {
+	t.Parallel()
+	testcases := []struct {
+		description     string
+		pathsToInsert   [][]int32
+		expectedIndices []int
+	}{
+		{
+			description: "single path",
+			pathsToInsert: [][]int32{
+				{8, 4, 9, 5},
+			},
+			expectedIndices: []int{0},
+		},
+		{
+			description: "insert ancestor after descendant",
+			pathsToInsert: [][]int32{
+				{8, 4, 9, 5},
+				{8, 4},
+			},
+			expectedIndices: []int{
+				1,
+				0,
+			},
+		},
+		{
+			description: "insert multiple ancestors after descendant",
+			pathsToInsert: [][]int32{
+				{20, 15, 10, 5}, // 0
+				{20},            // 1
+				{20, 10},        // 2
+				{20, 15},        // 3
+			},
+			expectedIndices: []int{
+				1,
+				2,
+				3,
+				0,
+			},
```

Comment:
Is the actual order of returned indices significant? Seems like we should just verify that expected indices equals [0..len(pathsToInsert)), so we could sort the return before comparing.